web/src/features/time/TimeController.tsx

```diff
@@ -104,25 +104,31 @@ export default function TimeController({ className }: { className?: string }) {
     (timeRange: TimeRange) => {
       // Set time slider to latest value before switching aggregate to avoid flickering
       setSelectedDatetime({
-        datetime: selectedDatetime.datetime,
+        datetime: datetimes.at(-1),
```

Comment:
The `onToggleGroupClick` was changed each time we changed the current time, causing a full re-render of the TimeController child elements. Given the comment above, it seems like the this change should give us the independence from currently selected time, and not hinder the UX.
In the end I'm not sure why changing time range should depend on the currently selected datetime.